rpc/lib/client/ws_client.go

```diff
@@ -254,6 +256,7 @@ func (c *WSClient) reconnect() error {
 			c.Logger.Error("failed to redial", "err", err)
 		} else {
 			c.Logger.Info("reconnected")
+			c.ReconnectCh <- true
```

Comment:
I'd prefer an optional callback here:

```
go c.OnReconnect()
```

Reply:
@melekes does this apply though since I believe that ReconnectCh was exported so that we could use it in `rpc/client/httpclient.go` at https://github.com/tendermint/tendermint/pull/689/files#diff-a969276db2d519208a2bf7aa9447acf6R338
<img width="502" alt="screen shot 2017-09-29 at 1 21 50 am" src="https://user-images.githubusercontent.com/4898263/31005146-b14a68a4-a4b4-11e7-94ed-6f0a275d7596.png">
as this file is in `rpc/lib/client` so we need that visibility to be able to use it there.

Reply:
You're true, for me it is the best way to delegate reconnection behavior to WSEvents.
I've just fixed the unit tests.

Reply:
@odeke-em yes, it is. you probably did not notice, but `OnReconnect` is also exported. so in httpclient we should be able to do:

```
w.ws.OnReconnect = func() {
  w.redoSubscriptions()
}
```

Reply:
Ooh I see, roger that @melekes.

Reply:
I rewrote some code here https://github.com/tendermint/tendermint/commit/da5bd254b6d124a6b4fbb8721f4e001f7e370682. Feel free to leave your comments.